ssl/quic/quic_port.c

```diff
@@ -136,11 +161,25 @@ static int port_init(QUIC_PORT *port)
     ossl_list_port_insert_tail(&port->engine->port_list, port);
     port->on_engine_list    = 1;
     port->bio_changed       = 1;
-    return 1;
 
+    /* Generate random key for token encryption */
+    if ((port->token_ctx = EVP_CIPHER_CTX_new()) == NULL
+        || (cipher = EVP_CIPHER_fetch(port->engine->libctx,
+                                      "AES-256-GCM", NULL)) == NULL
+        || !EVP_EncryptInit_ex(port->token_ctx, cipher, NULL, NULL, NULL)
+        || (key_len = EVP_CIPHER_CTX_get_key_length(port->token_ctx)) <= 0
+        || (token_key = OPENSSL_malloc(key_len)) == NULL
+        || !RAND_bytes_ex(port->engine->libctx, token_key, key_len, 0)
+        || !EVP_EncryptInit_ex(port->token_ctx, NULL, NULL, token_key, NULL))
+        goto err;
+
+    ret = 1;
 err:
-    port_cleanup(port);
-    return 0;
+    EVP_CIPHER_free(cipher);
```

Comment:
Do I understand right all `EVP_*Init_ex()` took their own references to cipher so it is safe to drop `cipher` here?

Reply:
Yes